src/unix/async.c

```diff
@@ -81,20 +82,32 @@ int uv_async_send(uv_async_t* handle) {
 
 /* Only call this from the event loop thread. */
 static int uv__async_spin(uv_async_t* handle) {
+  int i;
   int rc;
 
   for (;;) {
-    /* rc=0 -- handle is not pending.
-     * rc=1 -- handle is pending, other thread is still working with it.
-     * rc=2 -- handle is pending, other thread is done.
+    /* 997 is not completely chosen at random. It's a prime number, acyclical
+     * by nature, and should therefore hopefully dampen sympathetic resonance.
      */
```

Comment:
how about using a decay function to adjust (gradually decrease) the spin count, such that we yield more than we spin, if it takes more time to get the lock?

Reply:
Gireesh, can you post your suggestion in the shape of a patch or pseudo-code?

Reply:
@bnoordhuis - 

```patch
--- a/src/unix/async.c
+++ b/src/unix/async.c
@@ -85,11 +85,12 @@ static int uv__async_spin(uv_async_t* handle) {
   int i;
   int rc;
 
+  int c = 997;
   for (;;) {
     /* 997 is not completely chosen at random. It's a prime number, acyclical
      * by nature, and should therefore hopefully dampen sympathetic resonance.
      */
-    for (i = 0; i < 997; i++) {
+    for (i = 0; i < c; i++) {
       /* rc=0 -- handle is not pending.
        * rc=1 -- handle is pending, other thread is still working with it.
        * rc=2 -- handle is pending, other thread is done.
@@ -99,10 +100,13 @@ static int uv__async_spin(uv_async_t* handle) {
       if (rc != 1)
         return rc;
 
       /* Other thread is busy with this handle, spin until it's done. */
       cpu_relax();
     }
 
+    c = 0.9 * c + 1;
```

Reply:
Thanks, I understand what you mean now. How does it affect the benchmarks? You can run them like this:
```
$ for T in async{1,2,4,8}; do taskset 1 build/uv_run_benchmarks_a $T; done
```
(Try it with and without the `taskset 1`.)

Reply:
ah! @bnoordhuis , I did not test this. and now when I tried, realized that my build system is not up-to-date to build this, and I might not get time to fix it now. So feel free to ignore this; I shall come back with an addon PR, if / when I get time and benchmarks show something positive!

Reply:
@bnoordhuis If you intended to pin all threads of the benchmark process to cpu 1, the command should be:
`
$ for T in async{1,2,4,8}; do taskset -a -c 1 build/uv_run_benchmarks_a $T; done
`

Reply:
@gireeshpunathil I tried your suggestion just now but it doesn't seem to make a material difference.

Sorry @Parallel-Hao, missed your reply. `-a` is only necessary when changing the affinity of an existing process; it makes `taskctl` update the affinity of all threads instead of just the main thread.

Reply:
thanks @bnoordhuis for trying it! sorry for any noise I caused on this!